mcc/roosa/cmd/main.go

```diff
@@ -0,0 +1,152 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/route53"
+
+	"github.com/poka-yoke/spaceflight/mcc/ttl"
+)
+
+// Node type represents the reference between data.
+type Node struct {
+	parent   *Node
+	children []*Node
+	content  *route53.ResourceRecordSet
+}
+
+func (n *Node) isRoot() bool {
+	return n.parent == nil
+}
+
+var zoneName string
+
+// Init sets the flag parsing and input validations
+func Init() {
+	flag.StringVar(&zoneName, "zonename", "", "Hosted Zone's name to traverse")
+
+	flag.Parse()
+
+	if zoneName == "" {
+		log.Fatal("Insufficient input parameters!")
+	}
+}
+
+// PrintNode prints a Node, and all its children.
+func PrintNode(node *Node, indent int) {
+	indents := ""
+	for i := 0; i <= indent-1; i++ {
+		indents += "\t"
+	}
+	extra := ""
+	for i, record := range node.content.ResourceRecords {
+		extra += *record.Value
+		if i+1 < len(node.content.ResourceRecords) {
+			extra += ", "
+		}
+	}
+	fmt.Printf("%v%v %v %v\n", indents, *node.content.Name, *node.content.Type, extra)
+	indent++
+	for _, child := range node.children {
+		PrintNode(child, indent)
+	}
+}
+
+// FilterReferrals returns 'A', 'AAAA', and 'CNAME' records from a []*route53.ResourceRecordSet.
+func FilterReferrals(records []*route53.ResourceRecordSet) (filteredRecords []*route53.ResourceRecordSet) {
```

Comment:
If you don't name the output, you can just return the internal function's result.